src/scripts/auxLists/utils.ts

```diff
@@ -0,0 +1,233 @@
+  return `${prefix}${count ? ` top ${count}` : ''} on ${DISPLAY_CHAIN_NAMES[chain]}`
+}
+
+function getOutputPath(prefix: string, chainId: SupportedChainId): string {
+  return `src/public/${prefix}.${chainId}.json`
+}
+
+export function getLocalTokenList(listPath: string, defaultEmptyList: Partial<TokenList>): Partial<TokenList> {
+  try {
+    return JSON.parse(fs.readFileSync(listPath, 'utf8'))
+  } catch (error) {
+    console.warn(`Error reading token list from ${listPath}:`, error)
+    return defaultEmptyList
+  }
+}
+
+function getTokenListVersion(list: Partial<TokenList>, tokens: TokenInfo[]): TokenList['version'] {
+  const version = list.version || { major: 0, minor: 0, patch: 0 }
+  const listTokenAddresses = new Set(list.tokens?.map((token) => token.address.toLowerCase()) || [])
+  const tokensAddresses = new Set(tokens.map((token) => token.address.toLowerCase()))
+
+  // Check for removed tokens
+  if (
+    listTokenAddresses.size > tokensAddresses.size ||
+    Array.from(listTokenAddresses).some((address) => !tokensAddresses.has(address))
+  ) {
+    return { ...version, major: version.major + 1 }
+  }
+
+  // Check for added tokens
+  if (
+    listTokenAddresses.size < tokensAddresses.size ||
+    Array.from(tokensAddresses).some((address) => !listTokenAddresses.has(address))
+  ) {
+    return { ...version, minor: version.minor + 1 }
+  }
+
+  // Check for changes in token details
+  for (const listToken of list.tokens || []) {
+    const token = tokens.find((token) => token.address === listToken.address)
+    if (
+      token &&
+      (listToken.name !== token.name ||
+        listToken.symbol !== token.symbol ||
+        listToken.decimals !== token.decimals ||
+        listToken.logoURI !== token.logoURI)
+    ) {
+      return { ...version, patch: version.patch + 1 }
+    }
+  }
+
+  return version
+}
+
+export function saveUpdatedTokens({ chainId, prefix, logo, tokens, listName }: SaveUpdatedTokensParams): void {
+  const tokenListPath = path.join(getOutputPath(prefix, chainId))
+  const currentList = getLocalTokenList(tokenListPath, getEmptyList())
+  const updatedTokenList = { ...currentList, tokens, name: listName, logoURI: logo }
+
+  try {
+    const version = getTokenListVersion(currentList, tokens)
+    const updatedList: TokenList = { ...updatedTokenList, version, timestamp: new Date().toISOString() }
+    fs.writeFileSync(tokenListPath, JSON.stringify(updatedList, null, 2))
+    console.log(`Token list saved to ${tokenListPath}`)
+  } catch (error) {
+    console.error(`Error saving token list to ${tokenListPath}:`, error)
+  }
+}
+
+export async function processTokenList({
+  chainId,
+  tokens,
+  prefix,
+  logo,
+  logMessage,
+  shouldAddCountToListName = true,
+}: ProcessTokenListParams): Promise<void> {
+  const count = tokens.length
+  console.log(`🥇 ${logMessage} on chain ${chainId}`)
+
+  tokens.forEach((token, index) => {
+    const volumeStr = token.volume ? `: $${token.volume}` : ''
```

Comment:
nit (just suggestion): since you are improving this, u could add https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat, as the volume was a bit hard to read

Reply:
13a142f8223c